modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/RoleResourceTest.java

```diff
@@ -412,6 +451,14 @@ private long _getRoleId(Role role) {
 		return role.getId();
 	}
 
+	private Role _getRoleWithExternalReferenceCode() throws Exception {
+		Role role = _addRole(RoleConstants.TYPE_REGULAR);
+
+		role.setExternalReferenceCode(role.getName());
+
+		return role;
+	}
+
```

Comment:
I think this is not needed because the _addRole method I think already set the externalReferenceCode like the name. Please verify and in case just use it instead of this one

Reply:
It does not explicitly return that random Role, after that there is a line _addRole(role) which further manipulates the role given back by randomRole(). As a consequence the ERC has to be set again, otherwise it will be null. I have tested this. So it seems that that method _addRole() does not exactly works like one would expect.

Reply:
In summary _addRole() does not set the ERC although it may seem so.

Reply:
This could be resolved with either settings the ERC again in 
private Role _addRole(Role role)
or
private Role _addRole(int type)

Reply:
Alternatively, I could call randomRole() directly.